internal/painter/gl/draw.go

```diff
@@ -56,13 +56,21 @@ func (p *glPainter) rectCoords(size fyne.Size, pos fyne.Position, frame fyne.Siz
 }
 
 func (p *glPainter) drawTextureWithDetails(o fyne.CanvasObject, creator func(canvasObject fyne.CanvasObject) Texture,
-	pos fyne.Position, size, frame fyne.Size, fill canvas.ImageFill, alpha, aspect float32, pad int) {
+	pos fyne.Position, size, frame fyne.Size, fill canvas.ImageFill, alpha float32, pad int) {
 
 	texture := getTexture(o, creator)
 	if texture == NoTexture {
 		return
 	}
 
+	aspect := float32(0)
+	if img, ok := o.(*canvas.Image); ok {
+		aspect = painter.GetAspect(img)
+		if aspect == 0 {
+			fyne.LogError("Missing aspect ratio for loaded image", nil)
+			aspect = 1 // fallback, should not occur
+		}
+	}
```

Comment:
Should the zero check and fallback be moved outside of the type check and cast?
```suggestion
	if img, ok := o.(*canvas.Image); ok {
		aspect = painter.GetAspect(img)
	}
	if aspect == 0 {
		fyne.LogError("Missing aspect ratio for loaded image", nil)
		aspect = 1 // fallback, should not occur
	}
```

Reply:
Not in this case - see how 0 was passed in many times as a valid parameter.
0 is just not OK for images

Reply:
Ack